core/modules/file/file.entity.inc

```diff
@@ -122,7 +136,7 @@ class File extends Entity {
    */
   public function uri() {
     return array(
-      'path' => $this->uri,
+      'path' => 'file/' . $this->fid,
```

Comment:
This seems like it might have far-reaching implications. Definitely an API change that would need a change-record.